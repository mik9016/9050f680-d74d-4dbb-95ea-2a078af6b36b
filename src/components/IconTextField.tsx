import { TextField, InputAdornment } from '@mui/material'
import React, { useState, useCallback, useContext, useEffect } from 'react'
import { GlobalContext } from '../contexts/Context'

interface Props {
  iconStart: React.ReactElement
  label: string
}

export const IconTextField = ({
  iconStart,
  label,
}: Props): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  const [searchValue, setSearchValue] = useState('')

  useEffect(() => {
    globalCtx.setCurrentSearch(searchValue)
  }, [searchValue])

  const changeHandler = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>): void => {
      setSearchValue(e.currentTarget.value)
    },
    []
  )

  return (
    <TextField
      label={label}
      InputProps={{
        startAdornment: (
          <InputAdornment position="start">{iconStart}</InputAdornment>
        ),
      }}
      variant="outlined"
      onChange={changeHandler}
    />
  )
}
