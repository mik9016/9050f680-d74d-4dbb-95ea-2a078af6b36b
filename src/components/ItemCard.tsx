import {
  Card,
  CardActions,
  CardContent,
  CardMedia,
  IconButton,
  CardHeader,
  Button,
  Avatar,
  Typography,
} from '@mui/material'
import AddCircleIcon from '@mui/icons-material/AddCircle'
import React, { useContext } from 'react'
import LocationOnIcon from '@mui/icons-material/LocationOn'
import { GlobalContext } from '../contexts/Context'

export interface Props {
  bandName: string
  avatar: string
  picture: string
  venue: string
  venueName: string
  start: string
  end: string
}

const ItemCard = ({
  bandName,
  avatar,
  picture,
  venue,
  start,
  venueName,
  end,
  ref,
}: Props & { ref?: React.Ref<HTMLDivElement> }): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  const infos = {
    bandName: bandName,
    avatar: avatar,
    picture: picture,
    venue: venue,
    start: start,
    venueName: venueName,
    end: end,
  }
  const incrementCartNumber = () => {
    globalCtx.incrementNumber(infos)
  }
  const startDate = new Date(start)
  const endDate = new Date(end)

  const noPicUrl = 'https://picsum.photos/200/300'

  return (
    <Card
      ref={ref}
      sx={{
        width: { xs: 310, md: 345 },
        height: 400,
        margin: { xs: 0, md: 2 },
        my: { xs: 1 },
      }}
    >
      <CardHeader
        avatar={<Avatar alt="Avatar" src={avatar} />}
        title={bandName}
      />
      <CardMedia
        component="img"
        height="140"
        image={picture ? picture : noPicUrl}
        alt="picture"
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          <Button target="_blank" href={venue} rel="noopener">
            {' '}
            <LocationOnIcon color="primary" />
            {venueName}{' '}
          </Button>
        </Typography>
        <Typography
          gutterBottom
          variant="body2"
          color="text.secondary"
          component="div"
        >
          {`| Starts: ${startDate.getDate()}.${
            startDate.getMonth() + 1
          }.${startDate.getFullYear()},  ${startDate.getHours()}:${
            startDate.getMinutes() === 0 ? '00' : startDate.getMinutes()
          }`}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {`| Ends: ${endDate.getDate()}.${
            endDate.getMonth() + 1
          }.${endDate.getFullYear()},  ${endDate.getHours()}:${
            endDate.getMinutes() === 0 ? '00' : endDate.getMinutes()
          }`}
        </Typography>
      </CardContent>
      <CardActions>
        <IconButton aria-label="add" onClick={incrementCartNumber}>
          <AddCircleIcon color="primary" />
        </IconButton>
      </CardActions>
    </Card>
  )
}

export default ItemCard
