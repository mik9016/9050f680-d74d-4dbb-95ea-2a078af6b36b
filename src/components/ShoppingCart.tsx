import { Badge } from '@mui/material'
import React from 'react'
import ShoppingCartOutlinedIcon from '@mui/icons-material/ShoppingCartOutlined'

interface Props {
  number: number
}

export const ShoppingCart = ({ number }: Props): React.ReactElement => {
  return (
    <Badge badgeContent={number} color="error">
      <ShoppingCartOutlinedIcon color="action" />
    </Badge>
  )
}
