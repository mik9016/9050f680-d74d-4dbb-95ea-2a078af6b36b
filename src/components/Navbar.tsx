import React, { FC, ReactElement, useState, useContext } from 'react'
import { Link } from 'react-router-dom'
import { Box, Container, Toolbar } from '@mui/material'
import SearchIcon from '@mui/icons-material/Search'
import { GlobalContext } from '../contexts/Context'

import { IconTextField } from './IconTextField'
import { ShoppingCart } from './ShoppingCart'

const Navbar = (): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  return (
    <Box
      sx={{
        width: '100%',
        height: 'auto',
        padding: 1,
        paddingLeft: 3,
        backgroundColor: 'primary.main',
        position: 'fixed',
        zIndex: '4',
      }}
    >
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <IconTextField label="Search" iconStart={<SearchIcon />} />
          <Box sx={{ flexGrow: 1, display: 'flex' }}>
            <Box
              sx={{
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'flex-end',
                alignItems: 'center',
                marginLeft: 'auto',
                paddingLeft: '1rem',
              }}
            >
              <Link to={'/cart'}>
                <ShoppingCart number={globalCtx.currentCartNumber} />
              </Link>
            </Box>
          </Box>
        </Toolbar>
      </Container>
    </Box>
  )
}

export default Navbar
