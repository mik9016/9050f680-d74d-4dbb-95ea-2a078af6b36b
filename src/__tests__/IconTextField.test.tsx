import React from 'react'
import { act, create } from 'react-test-renderer';
import SearchIcon from '@mui/icons-material/Search'

import { IconTextField } from '../components/IconTextField';


test(`renders IconTextField without crashing`, async () => {
  const root = create(
    <IconTextField label="Search" iconStart={<SearchIcon />} />
  );

  expect(root.toJSON()).toMatchSnapshot();

  act(() => {
    root.unmount();
  });
});
