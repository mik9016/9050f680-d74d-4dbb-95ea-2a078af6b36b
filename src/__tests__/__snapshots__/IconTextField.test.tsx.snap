// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IconTextField without crashing 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    htmlFor=":r0:"
    id=":r0:-label"
  >
    Search
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        â€‹
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1in441m"
      >
        <span>
          Search
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
